src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1389,10 +1506,17 @@ public void Dispose()
                     list.ForEach(c => c._connection.Dispose());
                     list.Clear();
 
-                    if (_http2Connection != null)
+                    if (_http2Connections != null)
                     {
-                        _http2Connection.Dispose();
-                        _http2Connection = null;
+                        Http2Connection[] localHttp2Connections = _http2Connections;
+                        if (localHttp2Connections != null)
```

Comment:
This is all under the lock, so you shouldn't need to recheck for null, right? _http2Connections shouldn't be changing here.

Reply:
Seems local variable is not needed. Removed.